### ISSUE TITLE
Sinon included as dependency instead of devDependency

### ISSUE THREAD

**Issue by peternedap** [aws-amplify/amplify-js#9565]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Not applicable

### Amplify APIs

Not applicable

### Amplify Categories

storage

### Environment information

<details>

```
# Put output below this line

n/a
```

</details>

### Describe the bug

Amplify-js uses Sinon for testing, but includes it as a dependency rather than a devDependency. As a result, each project using `aws-amplify` will automatically include `sinon` as a subdependency:

```
yarn why v1.22.5
[1/4] Why do we have the module "sinon"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "sinon@7.5.0"
info Reasons this module exists
   - "aws-amplify#@aws-amplify#storage" depends on it
   - Hoisted from "aws-amplify#@aws-amplify#storage#sinon"
```

### Expected behavior

Sinon should be removed from the `dependencies` category and moved to the `devDependencies` in https://github.com/aws-amplify/amplify-js/blob/main/packages/storage/package.json#L51 

### Reproduction steps

either `yarn add aws-amplify` or `npm install aws-amplify` 

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by calebpollman** [comment]
Verified inclusion of `sinon` as a dependency of `@aws-amplify/storage`

**Comment by chrisbonifacio** [comment]
Closing this as it has been addressed by merging #9567
